### PR TITLE
[BE] Task#210871 Create new program faciltator

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/210871

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the facilitator management system by transitioning program facilitator creation to a more accessible POST endpoint.
  
- **New Features**
	- Implemented a secure method for adding program facilitators with necessary validations.

### CODE REVIEW

File: src/src/facilitator/facilitator.controller.ts [hunk 1]
```diff
@@ -196,4 +196,14 @@ export class FacilitatorController {
 	updateOkycResponse(@Req() req: any, @Body() body: any, @Res() res: any) {
 		return this.facilitatorService.updateOkycResponse(req, body, res);
 	}
+
+	@Post('/add/program-faciltator')
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/facilitator/facilitator.controller.ts:200]: rename as
`/program-facilitator/add`

File: src/src/facilitator/facilitator.controller.ts [hunk 2]
```diff
@@ -196,4 +196,14 @@ export class FacilitatorController {
...
+	@UseGuards(new AuthGuard())
+	createProgramFaciltator(
+		@Req() req: any,
+		@Body() body: any,
+		@Res() res: any,
+	) {
+		return this.facilitatorService.createProgramFaciltator(body, req, res);
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/facilitator/facilitator.controller.ts:207]: rename as `createProgramFacilitator`

File: src/src/facilitator/facilitator.service.ts [hunk 3]
```diff
@@ -2449,4 +2449,115 @@ export class FacilitatorService {
...
+		const ids = program_faciltators.map(
+			(facilitator) => facilitator.program_id,
+		);
+
+		if (!ids.includes(body?.program_id)) {
+			return res.json({
+				status: 422,
+				success: false,
+				data: {},
+				message: 'Cannot add faciltator for another program',
```
Comment (manojLondhe) [inline src/src/facilitator/facilitator.service.ts:2498]: 'Cannot add same user into multiple programs'

File: src/src/facilitator/facilitator.service.ts [hunk 4]
```diff
@@ -2449,4 +2449,115 @@ export class FacilitatorService {
...
+			});
+		}
+
+		//validation to check if the same faciltator data is present previously.
```
Comment (manojLondhe) [inline src/src/facilitator/facilitator.service.ts:2502]: Start code comments by giving one space after `//`
Start code comments in Capital case
eg:
		//start validation
=>
		// Start validations

File: src/src/facilitator/facilitator.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/facilitator/facilitator.controller.ts:200]: resolved

File: src/src/facilitator/facilitator.controller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/facilitator/facilitator.controller.ts:207]: resolved

**Pull request merged**